doc/nrf/ug_zigbee_adding_clusters.rst

```diff
@@ -0,0 +1,519 @@
+
+Adding ZCL clusters to application consists of expanding the Zigbee template sample with new application clusters.
+By default, the template sample includes only mandatory Zigbee clusters, that is the Basic and Identify clusters, required to identify a device within a Zigbee network.
+
+Cluster is a representation of a single functionality within a Zigbee node.
+Each cluster contains attributes that are stored in the device's memory and commands that can be used to modify or read the state of the device, including the cluster attributes.
+Clusters appropriate for a single device type such as a sensor or a light bulb are organized into an addressable container that is called an endpoint.
+For more information about the ZCL terminology, see `Common ZCL terms and definitions`_ in the ZBOSS stack user guide.
+
+An application can implement appropriate callback functions to be informed about specific cluster state changes.
+These functions can be used to alter device's behavior when the state of a cluster is changing as a result of some external event.
+
+Read the following sections for detailed steps about how to expand the Zigbee template sample.
+
+.. _ug_zigee_adding_clusters_requirements:
+
+Requirements
+************
+
+To take advantage of this guide, you need to be familiar with :ref:`ug_zigbee_architectures` and :ref:`ug_zigbee_configuring`, and have built and tested at least one of the available :ref:`zigbee_samples`.
+
+To test the functionalities implemented in this user guide, you need the following hardware and software:
+
+* One compatible development kit for programming the :ref:`Zigbee template <zigbee_template_sample>` sample
+* One compatible development kit for programming the :ref:`Zigbee network coordinator <zigbee_network_coordinator_sample>` sample
+
+You can also optionally use `nRF Sniffer for 802.15.4`_ configured for capturing Zigbee packets with Wireshark (see `Configuring Wireshark for Zigbee`_) to gather some of the required information, such as the short address of a Zigbee node.
+However, this guide does not describe Wireshark usage.
+
+.. _ug_zigee_adding_clusters_copying_template:
+
+.. rst-class:: numbered-step
+
+Copy Zigbee template sample
+***************************
+
+Use the :ref:`Zigbee template <zigbee_template_sample>` sample as the base for adding new ZCL clusters:
+
+1. Make sure that you meet the requirements for building the sample.
+#. Build and test the sample as described on its documentation page.
+#. Copy the contents of the :file:`samples/zigbee/template` directory to a new directory meant for your custom application.
+   For example, :file:`samples/zigbee/sensor`.
+
+.. _ug_zigee_adding_clusters_defining_role:
+
+.. rst-class:: numbered-step
+
+Define the device role
+**********************
+
+The Zigbee template sample by default implements the Zigbee Router role.
+To change the network role, complete the following steps:
+
+1. Lorem ipsum
+#. sit etiam dolorem amat
```

Comment:
@FilipZajdel , this was requested by Krzysztof Loska at the latest sprint review.

Reply:
Section removed, will be covered in https://projecttools.nordicsemi.no/jira/browse/KRKNWK-11394